### ISSUE TITLE
What is the release workflow for reactstrap? Can anybody starts this process?

### ISSUE THREAD

**Issue by juanmaguitar** [reactstrap/reactstrap#1101]
In the README there are some instructions about [how to do a release](https://github.com/reactstrap/reactstrap#releasing)

But if i launch the first command (with the idea of opening a PR with a new release) I get this...

```
⬢  reactstrap  master ⦾ ./scripts/release minor
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
npm WARN bootstrap@4.1.1 requires a peer of jquery@1.9.1 - 3 but none was installed.
npm WARN file-loader@1.1.5 requires a peer of webpack@^2.0.0 || ^3.0.0 but none was installed.
Environment variable GITHUB_TOKEN does not exist. Stopping deploy.
                                                                  %
```

So, the instructions imply that anyone can open a PR requesting a new release, but it seems only the ones w/ this `GITHUB_TOKEN` can do that

So, I have a lot of doubts about the workflow regarding this:
- Who's in charge of releasing a new version?
- How can we request a new release?
- Can anybody open a PR requesting a new release? What is this `GITHUB_TOKEN` for?
- In whatever the case is, what are be the steps to follow from someone requests a new release till the release is deployed to NPM package?

**Comment by supergibbs** [comment]
@TheSharpieOne Can speak to it more but like most open source projects, not everyone can release the package; that would cause havoc. You are welcome to request a new release if there are unreleased features/bug fixes that have been merged in. You can also build it locally for your project or even fork it and deploy to a new package name if you're in a big hurry.

**Comment by TheSharpieOne** [comment]
IMHO, I would love for travis to build and release a new version of reactstrap with every commit to master. It can use conventional commits to determine the version bump and automatically bump the package.json add the changes to the changelog and commit it, tag it, push it back to github and publish to npm.
I have been slacking when it comes to cutting releases as I am often using github to review PRs and merge them and such from phone and not from somewhere where I can pull down the changes and run the release stuff.

**Comment by juanmaguitar** [comment]
@TheSharpieOne, that would be great. I can help you with that if you want.
For now 👉 I'd like to request a new release with latest changes. 

How should I proceed (me or anyone) taking into account the error I'm getting (as explained) above when trying to create a PR to request the release → `./scripts/release minor`

Maybe a better explanation in the documentation regarding this would be helpful for the collaborators.

**Comment by TheSharpieOne** [comment]
the release script (IIRC) can already determine the version bump, it just a matter of get travis to run it and as well as commit and push the changes and publish to npm as well as not get caught in a loop with the commits it makes.

**Comment by juanmaguitar** [comment]
Great! Anyway, count on my support if you need it
For now... When do you think you'll be able to release the next version w/ latest changes?

**Comment by juanmaguitar** [comment]
Oks, In Mac the issue I was having when trying to create the PR for the release was solved by providing a Github Token (that can be get [here](https://github.com/settings/tokens))

```
GITHUB_TOKEN=027bc59b64c1294347f2b6842e8c67b33f769557 ./scripts/release minor
```

This process will add changes to be included in released version to `CHANGELOG` and the proper version to `package.json`

With these changes, a new PR ([like this one](https://github.com/reactstrap/reactstrap/pull/1105)) can be submitted

**Comment by TheSharpieOne** [comment]
You don't need to specify `minor` at the end of that command, it will automatically determine the bump according to commits since the last release and semver. Specifying the bump type overrides that and may lead to a no semver conforming release version.

I made a change to the travis config which should hopefully generate the dist assets and publish them to npm upon a new tag being pushed to the repo. This should mean that we can now have PRs similar to the one you created and after it gets merged, simply tag that commit and it will be published.

**Comment by juanmaguitar** [comment]
Oks, then the documentation regarding this should be modified en the `README.md` because [now it says to specify the release type (major, minor or patch)](https://github.com/reactstrap/reactstrap#create-release-branch) → `./scripts/release <release-type>`

**Comment by TheSharpieOne** [comment]
I updated that section of the readme to denote that is optional as well as who can run what parts and that travis is the one publishing to npm

**Comment by TheSharpieOne** [comment]
Sorry for the double post but I just wanted to say:
In the end, it's not quite how I had hoped it would work, there is still the manual step of generating the changelog and creating that PR as well as tagging it once merged... but at least now someone without write access to the repo can initiate a release and someone with write access to the repo can merge it and tag it to trigger it to be published. It's at least one step closer and makes it so that I can publish a release from my phone (by simply tagging it; given that someone else has opened a PR like you (@juanmaguitar) did.)

Maybe eventually we will get it to the point where travis does everything automatically with every commit that passes the build/tests so that no one is ever waiting on something to be released (sure we would get to version numbers like `6.54.21`, but it's semver and most people use `^`)
